starlette/routing.py

```diff
@@ -100,7 +100,10 @@ def replace_params(
 
 
 # Match parameters in URL paths, eg. '{param}', and '{param:int}'
-PARAM_REGEX = re.compile("{([a-zA-Z_][a-zA-Z0-9_]*)(:[a-zA-Z_][a-zA-Z0-9_]*)?}")
+PARAM_REGEX = re.compile(
+    "{([a-zA-Z_][a-zA-Z0-9_]*)(:[a-zA-Z_][a-zA-Z0-9_]*)?"
+    "((?:(?:\\(.*\\))|(?:\\[.*\\]))*)}"
```

Comment:
This regex is *wwwaaaayyyy* to complex for us to have in without comments explaining it.

Reply:
I tried to make it more simple but it doesn’t work

Reply:
He meant to add a comment on the code `# This regex matches...`

Reply:
I understand maybe I over complicate the pattern I will check it again

Reply:
I made it much simpler can you please check it again ?